doc_internal/support-tiers.md

```diff
@@ -0,0 +1,119 @@
+
+## The Tiers
+
+1. Working git at all times - can you expect a random checkout to be usable on this particular platform/terminal?
+   If this is broken, we should revert the offending commits ASAP.
+2. Tested before release, we don't want to ship with it broken
+3. We hope this works and will try to fix it if it doesn't, but no guarantees
+4. We don't care. If it works, that's cool.
+
+These tiers depend on how important a given target is to us, and how much information we get about it.
+
+That means, a tier 1 target requires a core developer to run it in anger, and working CI for operating systems/architectures.
+Since we don't have automated testing of terminals, CI isn't required there.
+
+A tier 2 target needs us to invest a conscious effort to test it before release, but of course if we don't run it ourselves we may not catch things
+that aren't covered in tests. Users can help here. So a tier 2 target is better the more it is used. Note: We may decide to skip some tests for patch releases.
+
+A tier 3 target is best effort. These can move up if they become more important, or someone on the core team uses and maintains it.
+Or it can move down if it conflicts with improving other targets.
+
+A tier 4 target is one that we aren't interested in, bug reports will most likely be closed, but we may invest some time trying to help.
+If this works often enough it may be promoted to tier 3. We may also reject patches if they are too invasive.
+
+The distinction between tier 1 and 2 is really just an expression of our confidence that any random git checkout or release will work.
+The distinction between tier 2 and 3 is much more important - these aren't supported beyond "we'll try if you tell us".
+And tier 4 is just a list of platforms we know fish doesn't work on.
+
+## The current targets: Operating Systems / Architectures
+
+Unless anything else is mentioned, these refer to current mainstream versions.
+
+Tier 1:
+
+- Linux, with current glibc, on x86_64
+- macOS, aarch64
+
+- WSL (as "basically glibc linux" with some use among the core developers)
+
+Tier 2+ (have CI but not enough use among core devs):
+
+- Linux, musl, x86_64
+- Linux, older glibc, x86_64
+- FreeBSD
+
+Tier 2:
+
+- NetBSD, x86_64
+- Linux, glibc, aarch64?
+- macOS x86_64?
+
+Tier 3:
+
+- OpenBSD
+- OpenSolaris/Illumos/OpenIndiana?
+- Architectures other than x86_64 and aarch64 (and possibly i686)
+
+Tier 4:
+- Architectures and operating systems not supported by rust.
+- Cygwin/Msys
+- Haiku?
+- [...]
+- [...]
+- [Long scroll, look at your watch or phone]
+- [...]
+- Windows native
+- Nonstop
+
+## The current targets: Terminals
+
+Tier 1:
+
+- Konsole
+- MS Terminal
+- tmux (used in our CI)
+- other terms the core devs use
+
+Tier 2:
+
+- Ghostty
+- Wezterm
+- Alacritty
+- Kitty?
+- xterm?
+- iTerm?
+- Terminal.app?
+- Gnome Terminal / Console?
```

Comment:
I'd definitely add VSCode console, maybe tier 2, I think that's an important one (with a lot of potential for problems unfortunately)